api/s2n.h

```diff
@@ -241,12 +241,32 @@ S2N_API extern int s2n_init(void);
 S2N_API extern int s2n_cleanup(void);
 
 /**
- * Create a new s2n_config object. This object can (and should) be associated with many connection objects.
+ * Creates a new s2n_config object and initializes it with default system certificates. This object
+ * can (and should) be associated with many connection objects.
  *
- * @returns returns a new configuration object suitable for associating certs and keys. 
+ * The returned config should be freed with `s2n_config_free()` after it's no longer in use by any
+ * connection.
+ *
+ * @returns A new configuration object suitable for configuring connections and associating certs
+ * and keys.
```

Comment:
I think it's worth another note here that this will initialize with default system certs. We should probably assume this is the first config API people will look at.

Reply:
Nope, I just completely missed the note :)